scheduler/pkg/scheduler/scheduler.go

```diff
@@ -194,19 +201,22 @@ func (s *SimpleScheduler) scheduleToServer(modelName string) error {
 		}
 	}
 
-	if !ok && !okWithMinReplicas {
-		msg := "Failed to schedule model as no matching server had enough suitable replicas"
-		logger.Debug(msg)
-		// we do not want to reset the server if it has live replicas or loading replicas
-		// in the case of loading replicas, we need to make sure that we can unload them later.
-		// for example in the case that a model is just marked as loading on a particular server replica
-		// then it gets a delete request (before it is marked as loaded or available) we need to make sure
-		// that we can unload it from the server
-		s.store.FailedScheduling(latestModel, msg, !latestModel.HasLiveReplicas() && !latestModel.IsLoadingOrLoadedOnServer())
-		return errors.New(msg)
+	if !ok {
+		s.serverScaleUp(latestModel)
```

Comment:
we try not to publish to the eventhub while holding a lock (as it can deadlock). I suggest to return the msg and send the event in the caller instead. This also follows the same pattern elsewhere.

Reply:
Ah yes, forgot about that.